libs/os/dir.cpp

```diff
@@ -0,0 +1,40 @@
+#include "dir.h"
+
+#include <QDir>
+#include <QString>
+#include <QStringList>
+#include <QFileInfo>
+
+Directory::Directory(const char* name) {
+    this->dir = new QDir(name);
+}
+
+bool Directory::good() {
+    if(this->dir) {
+        QDir* dir_instance = reinterpret_cast<QDir*>(this->dir);
+        return dir_instance->exists();
+    }
+    return false;
+}
+
+void Directory::close() {
+    if(this->dir) {
+        QDir* dir_instance = reinterpret_cast<QDir*>(this->dir);
+        delete dir_instance;
+        this->dir = nullptr;
+    }
+}
+
+const char* Directory::read_and_increment(){
+    if(this->dir) {
+        QDir* dir_instance = reinterpret_cast<QDir*>(this->dir);
+        QStringList entryList = dir_instance->entryList(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot);
+        if(this->entry_idx < entryList.size()) {
+            QString file_path = entryList.at(this->entry_idx);
+            QString file_name = QFileInfo(file_path).fileName();
+            this->entry_idx++;
+            return file_name.toStdString().c_str();
```

Comment:
.toLatin1() is more correct for this sort of conversions.
And we are returning pointer to temp local variable here ⚰️

Reply:
changed

Reply:
The reason why glib was removed is because qt provides directory management therefore glib seems to be redundant.

Reply:
I would love to know your take on this, this project already heavily uses qt why not use qt core for internal stuff as well.

Reply:
Qt is used almost exclusively for UI.
Ic no positive effects of this change. It's not even removing glib dependency, which is used in other places.
While it adds dependency on big juicy framework, which may be unwelcome in case of some potential porting. Imagine doing Gtk->Qt port if not just UI was dependent on Gtk.
Also performance and behavior compatibility of this change are questionable.